### ISSUE TITLE
submissions: 500 Internal Server Error

### ISSUE THREAD

**Issue by staeglis** [DOMjudge/domjudge#1028]
### Description of the problem
> 500 Internal Server Error when visiting jury/submissions. The issue occured after the upgrade from Domjudge 7.2.0 und Ubuntu 18.04 to Domjudge 7.3.2 and Ubuntu 20.04

### Your environment
> Include details about your installation here.
> - DOMjudge version: 7.3.2
> - Operating system / Linux distribution and version: Ubuntu 20.04
> - Webserver (e.g. Apache or nginx)

### Steps to reproduce
> Replace this with a description how we can reproduce your bug.
> - Visit the domjudge installation web page
> - Login as admin
> - Got to jury/submissions

### Expected behaviour
> Submissions are shown

### Actual behaviour
> 500 Internal Server Error

```
An exception has been thrown during the rendering of a template ("An exception occurred while executing 'SELECT r.runresult, t.rank, t.description FROM testcase t LEFT JOIN judging_run r ON (r.testcaseid = t.testcaseid AND r.judgingid = ?) WHERE t.probid = ? ORDER BY rank' with params [328, 1]: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 5").
```

### Any other information that you want to share?
> Please include webserver, symfony and judgedaemon log snippets here as appropriate.
> Screenshots may help in case of UI bugs.

**Comment by nickygerritsen** [comment]
I think you are hitting https://github.com/DOMjudge/domjudge/commit/eb961bd0026bbf003330fb60e8c5828f2cc3678f#diff-513f2059a1f2338bfd7b94e63707e8e6c6332957d63f8df9f70c8035d6d32d31

However 7.3.2 should have a fix for that. Can you remove the contents of `webapp/var/cache/prod` and also make sure you have upgraded the database?

**Comment by staeglis** [comment]
```
root@domjudge:/opt/domjudge/domserver/webapp/var/cache/prod# ls -l   
total 408
-rw-rwsr--+  1 www-data www-data   9770 Jan 25 13:41 annotations.map
drwxrwsr-x+  2 www-data www-data  28672 Jan 25 13:41 ContainerHr4yuRI
-rw-rwsr--+  1 www-data www-data  15645 Jan 25 13:48 djDbConfig.php
-rw-rwsr--+  1 www-data www-data    182 Jan 25 13:48 djDbConfig.php.meta
drwxrwsr-x+  3 www-data www-data   4096 Jan 25 13:41 doctrine
drwxrwsr-x+  2 www-data www-data   4096 Jan 26 15:28 jms_serializer
drwxrwsr-x+  8 www-data www-data   4096 Jan 26 15:15 pools
-rw-rwsr--+  1 www-data www-data    761 Jan 25 13:41 srcApp_KernelProdContainer.php
-rw-rwsr--+  1 www-data www-data      0 Jan 25 13:41 srcApp_KernelProdContainer.php.lock
-rw-rwsr--+  1 www-data www-data 103815 Jan 25 13:41 srcApp_KernelProdContainer.php.meta
-rw-rwsr--+  1 www-data www-data  35642 Jan 25 13:41 srcApp_KernelProdContainer.preload.php
drwxrwsr-x+ 60 www-data www-data   4096 Jan 27 07:28 twig
-rw-rwsr--+  1 www-data www-data  64880 Jan 26 15:15 UrlGenerator.php
-rw-rwsr--+  1 www-data www-data  14020 Jan 26 15:15 UrlGenerator.php.meta
-rw-rwsr--+  1 www-data www-data  72465 Jan 25 13:41 UrlMatcher.php
-rw-rwsr--+  1 www-data www-data  14020 Jan 25 13:41 UrlMatcher.php.meta
root@domjudge:/opt/domjudge/domserver/webapp/var/cache/prod# rm -r *
root@domjudge:/opt/domjudge/domserver/webapp/var/cache/prod# ls -l
total 412
-rw-rw-r--+  1 www-data www-data   9770 Mar 29 14:08 annotations.map
drwxrwsr-x+  2 www-data www-data  28672 Mar 29 14:08 ContainerHr4yuRI
-rw-r--r--+  1 www-data www-data  15645 Mar 29 14:08 djDbConfig.php
-rw-r--r--+  1 www-data www-data    182 Mar 29 14:08 djDbConfig.php.meta
drwxrwsr-x+  3 www-data www-data   4096 Mar 29 14:08 doctrine
drwxrwsr-x+  2 www-data www-data   4096 Mar 29 14:08 jms_serializer
drwxrwsr-x+  8 www-data www-data   4096 Mar 29 14:08 pools
-rw-r--r--+  1 www-data www-data    761 Mar 29 14:08 srcApp_KernelProdContainer.php
-rw-rw-r--+  1 www-data www-data      0 Mar 29 14:08 srcApp_KernelProdContainer.php.lock
-rw-r--r--+  1 www-data www-data 103815 Mar 29 14:08 srcApp_KernelProdContainer.php.meta
-rw-r--r--+  1 www-data www-data  35642 Mar 29 14:08 srcApp_KernelProdContainer.preload.php
drwxrwsr-x+ 18 www-data www-data   4096 Mar 29 14:09 twig
-rw-r--r--+  1 www-data www-data  64880 Mar 29 14:08 UrlGenerator.php
-rw-r--r--+  1 www-data www-data  14020 Mar 29 14:08 UrlGenerator.php.meta
-rw-r--r--+  1 www-data www-data  72465 Mar 29 14:08 UrlMatcher.php
-rw-r--r--+  1 www-data www-data  14020 Mar 29 14:08 UrlMatcher.php.meta
root@domjudge:/opt/domjudge/domserver/webapp/var/cache/prod# chmod 777 ^C
root@domjudge:/opt/domjudge/domserver/webapp/var/cache/prod# cd ..
root@domjudge:/opt/domjudge/domserver/webapp/var/cache# chmod -R 777 ./
root@domjudge:/opt/domjudge/domserver/webapp/var/cache# /opt/domjudge/domserver/bin/dj_setup_database upgrade
Database credentials read from '/opt/domjudge/domserver/etc/dbpasswords.secret'.
                                                              
                    Application Migrations                    
                                                              

WARNING! You have 2 previously executed migrations in the database that are not registered migrations.
    >> 2019-08-03 15:08:40 (20190803150840)
    >> 2019-11-18 00:00:00 (20191118000000)
No migrations to execute.
DOMjudge database upgrade completed.
```

The still still remains.

**Comment by nickygerritsen** [comment]
OK found it, can you change line 340 of `webapp/src/Twig/TwigExtension.php` to read
```sql
SELECT er.result as runresult, t.`rank`, t.description
```
I.e. add ` around rank? I think that will work. If so I will push a fix for that for the next 7.3 release.

**Comment by staeglis** [comment]
This doesn't fix the issue. But changing line 358f from
```
                'SELECT r.runresult, t.rank, t.description
                  FROM testcase t
                  LEFT JOIN judging_run r ON (r.testcaseid = t.testcaseid
                                              AND r.judgingid = :judgingid)
                  WHERE t.probid = :probid ORDER BY rank',
                [':judgingid' => $judgingId, ':probid' => $probId]);
```
to
```
                'SELECT r.runresult, t.rank, t.description
                  FROM testcase t
                  LEFT JOIN judging_run r ON (r.testcaseid = t.testcaseid
                                              AND r.judgingid = :judgingid)
                  WHERE t.probid = :probid ORDER BY t.rank',
                [':judgingid' => $judgingId, ':probid' => $probId]);
```
does (ORDER BY t.rank). At least there is no error messages anymore and the output looks reasonable :)

**Comment by nickygerritsen** [comment]
OK that also makes sense, thanks!

**Comment by nickygerritsen** [comment]
Fixed in 296a215f
